plan/plan.go

```diff
@@ -154,6 +189,63 @@ func (t *planTable) newPlanKey(e *endpoint.Endpoint) planKey {
 	return key
 }
 
+// Return an iterator of changes to resource records sets
+func (c *Changes) All() iter.Seq[*RRSetChange] {
+	return func(yield func(change *RRSetChange) bool) {
+		rrSetCreates := map[RRKey]*RRSetChange{}
+		rrSetDeletes := map[RRKey]*RRSetChange{}
+		rrSetUpdates := map[RRKey]*RRSetChange{}
+		for _, action := range []struct {
+			endpoints    *[]*endpoint.Endpoint
+			rrSetChanges *map[RRKey]*RRSetChange
+		}{
+			{endpoints: &c.UpdateNew, rrSetChanges: &rrSetUpdates},
+			{endpoints: &c.UpdateOld, rrSetChanges: &rrSetUpdates},
+			{endpoints: &c.Create, rrSetChanges: &rrSetCreates},
+			{endpoints: &c.Delete, rrSetChanges: &rrSetDeletes},
+		} {
```

Comment:
```suggestion
        actions := []struct {
			endpoints    *[]*endpoint.Endpoint
			rrSetChanges *map[RRKey]*RRSetChange
		}{
			{endpoints: &c.UpdateNew, rrSetChanges: &rrSetUpdates},
			{endpoints: &c.UpdateOld, rrSetChanges: &rrSetUpdates},
			{endpoints: &c.Create, rrSetChanges: &rrSetCreates},
			{endpoints: &c.Delete, rrSetChanges: &rrSetDeletes},
		}
		for _, action := actions {
```
I suggest to write something like this in order improve code readability. Wdyt ?

Reply:
I don't know; I don't consider myself to be a Go developer - this is the pretty much the first Go code I've written - so if you consider your proposed form to be easier to read/understand then I'm happy to be guided.

I proposed the original form predominantly in the interests of eliminating the additional named variable (i.e. `actions`). I'm creating the array of anonymous structures purely for the convenience of `for` loop. When one has an array of anonymous structures, one might as well have an anonymous array of anonymous structures?

Reply:
In go, there is a focus on code readability. 
So yes, would you please update this PR accordingly ?

Reply:
Addressed in https://github.com/kubernetes-sigs/external-dns/pull/4928/commits/42b90683d93b2ba725d1eb4d578775ec65ee25e5.
> In go, there is a focus on code readability.

It's probably a reasonable assumption that most programming languages, and developers, strive for readability; the challenge is that it's very subjective.